analyzers/tests/SonarAnalyzer.UnitTest/Extensions/KnownReferenceTests.cs

```diff
@@ -0,0 +1,295 @@
+        compilation.SetupGet(x => x.ReferencedAssemblyNames).Returns(new[] { identity });
+        var sut = new KnownReference(PublicKeyTokenIs("c5b62af9de6d7244"));
+        sut.IsReferencedBy(compilation.Object).Should().BeFalse();
+    }
+
+    [TestMethod]
+    public void OptionalPublicKeyTokenIs_SucceedsWhenKeyIsMissing()
+    {
+        var compilation = new Mock<Compilation>("compilationName", ImmutableArray<MetadataReference>.Empty, new Dictionary<string, string>(), false, null, null);
+        var identity = new AssemblyIdentity("assemblyName", hasPublicKey: false);
+        compilation.SetupGet(x => x.ReferencedAssemblyNames).Returns(new[] { identity });
+        var sut = new KnownReference(OptionalPublicKeyTokenIs("c5b62af9de6d7244"));
+        sut.IsReferencedBy(compilation.Object).Should().BeTrue();
+    }
+
+    [DataTestMethod]
+    [DataRow("Test", "1.0.0.0", false)]
+    [DataRow("Test", "1.9.9.99", false)]
+    [DataRow("TestMy", "2.0.0.0", true)]
+    [DataRow("MyTest", "2.0.0.0", false)]
+    [DataRow("TestMy", "3.5.0.0", true)]
+    [DataRow("TestMy", "3.5.0.1", false)]
+    [DataRow("Test", "10.0.0.0", false)]
+    public void Combinator_NameStartWith_Test_And_Version_Between_2_0_And_3_5(string name, string version, bool expected)
+    {
+        var sut = new KnownReference(StartsWith("Test").And(VersionBetween(new Version(2, 0, 0, 0), new Version(3, 5, 0, 0))));
+        var compilation = new Mock<Compilation>("compilationName", ImmutableArray<MetadataReference>.Empty, new Dictionary<string, string>(), false, null, null);
+        var identity = new AssemblyIdentity(name, new Version(version));
+        compilation.SetupGet(x => x.ReferencedAssemblyNames).Returns(new[] { identity });
+        sut.IsReferencedBy(compilation.Object).Should().Be(expected);
+    }
+
+    [DataTestMethod]
+    [DataRow("Start", true)]
+    [DataRow("End", true)]
+    [DataRow("StartOrEnd", true)]
+    [DataRow("StartTest", true)]
+    [DataRow("TestEnd", true)]
+    [DataRow("EndStart", false)]
+    [DataRow("EndSomething", false)]
+    [DataRow("SomethingStart", false)]
+    public void Combinator_StartsWith_Start_Or_EndsWith_End(string name, bool expected)
+    {
+        var sut = new KnownReference(StartsWith("Start"), EndsWith("End"));
+        var compilation = new Mock<Compilation>("compilationName", ImmutableArray<MetadataReference>.Empty, new Dictionary<string, string>(), false, null, null);
+        var identity = new AssemblyIdentity(name);
+        compilation.SetupGet(x => x.ReferencedAssemblyNames).Returns(new[] { identity });
+        sut.IsReferencedBy(compilation.Object).Should().Be(expected);
+    }
+
+    [TestMethod]
+    public void XUnitAssert_2_4()
+    {
+        var compilation = SolutionBuilder.Create()
+            .AddProject(AnalyzerLanguage.CSharp)
+            .AddReferences(NuGetMetadataReference.XunitFramework("2.4.2"))
+            .AddSnippet("// Empty file")
+            .GetCompilation();
```

Comment:
There's a helper for this
```suggestion
        var compilation = TestHelper.CompileCS("// Empty file", NuGetMetadataReference.XunitFramework("2.4.2")).Model.Compilation
```